### PR TITLE
Refresh MAPPING list from Rails

### LINKED ISSUE #84: Duplicated key's in timezones.rb
There are many duplicated keys in timezones.rb. For example lines 5 and 6 have the same key. In Ruby 2.2.0 this will produce a deprecation warning. Is this able to be fixed? I am willing to do the fix also I am just not sure which one to take since they have the same key?

```
lib/harvest/timezones.rb:5: warning: duplicated key at line 6 ignored: "pacific/midway"
lib/harvest/timezones.rb:18: warning: duplicated key at line 19 ignored: "america/mexico_city"
lib/harvest/timezones.rb:25: warning: duplicated key at line 26 ignored: "america/lima"
lib/harvest/timezones.rb:39: warning: duplicated key at line 40 ignored: "europe/dublin"
lib/harvest/timezones.rb:60: warning: duplicated key at line 61 ignored: "europe/berlin"
lib/harvest/timezones.rb:80: warning: duplicated key at line 81 ignored: "europe/moscow"
lib/harvest/timezones.rb:80: warning: duplicated key at line 82 ignored: "europe/moscow"
lib/harvest/timezones.rb:88: warning: duplicated key at line 89 ignored: "asia/muscat"
lib/harvest/timezones.rb:95: warning: duplicated key at line 96 ignored: "asia/karachi"
lib/harvest/timezones.rb:98: warning: duplicated key at line 99 ignored: "asia/kolkata"
lib/harvest/timezones.rb:98: warning: duplicated key at line 100 ignored: "asia/kolkata"
lib/harvest/timezones.rb:98: warning: duplicated key at line 101 ignored: "asia/kolkata"
lib/harvest/timezones.rb:103: warning: duplicated key at line 104 ignored: "asia/dhaka"
lib/harvest/timezones.rb:109: warning: duplicated key at line 110 ignored: "asia/bangkok"
lib/harvest/timezones.rb:124: warning: duplicated key at line 125 ignored: "asia/tokyo"
lib/harvest/timezones.rb:124: warning: duplicated key at line 126 ignored: "asia/tokyo"
lib/harvest/timezones.rb:130: warning: duplicated key at line 131 ignored: "australia/melbourne"
lib/harvest/timezones.rb:138: warning: duplicated key at line 139 ignored: "asia/magadan"
lib/harvest/timezones.rb:144: warning: duplicated key at line 145 ignored: "pacific/auckland"
```

### PR DESCRIPTION
**Pull request by parndt**
@zmoazeni this fixes #84 and it also reverses the keys and values in the MAPPING list.  What do you think will happen if we do this?

### CODE REVIEW

**Comment by jvdp** [comment]
Because this just reverses the mapping but doesn't change the code that uses it, I think it would break timezone resolution. For another approach see #86.

**Comment by parndt** [comment]
@jvdp while true that #86 doesn't break anything, that version also creates a custom, diverged copy of the mappings from what ActiveSupport has.. I think that the best way forward is to use the mappings from ActiveSupport but change the code that does timezone resolution.

For backward compatibility, if we desire it, we can reverse the finder to mimic the original behaviour.

@zmoazeni which approach would you prefer in your codebase?

**Comment by zmoazeni** [comment]
#86 is probably the immediate fix, but I think what really needs to happen is to just drop the `Timezones` module altogether along with the [code that uses it](https://github.com/zmoazeni/harvested/blob/41c5a2e595df9ca4d9828a5e04d37a48976c0cfa/lib/harvest/user.rb#L42-L55).

I'm not 100% sure, but I think this PR as written will break down because [user currently expects certain keys](https://github.com/zmoazeni/harvested/blob/41c5a2e595df9ca4d9828a5e04d37a48976c0cfa/lib/harvest/user.rb#L50).

I'm fine with a backwards incompatible update and starting the v4 line.

**Comment by parndt** [comment]
Sorry, this PR was half a patch.. I've updated the PR with the rest of the code to find the correct value by value.

See http://www.ruby-doc.org/core-2.1.5/Hash.html#method-i-key for details on `Hash#key`

@zmoazeni do you have a testing environment set up for this already? Could you test this? If you're unable, I'll guess I'll go and register a temporary harvest account for this purpose!

File: lib/harvest/user.rb [hunk 1]
```diff
@@ -47,10 +47,12 @@ def timezone=(timezone)
       when 'mst', 'mdt' then self.timezone = 'america/denver'
       when 'pst', 'pdt' then self.timezone = 'america/los_angeles'
       else
-        if Harvest::Timezones::MAPPING[tz]
-          self["timezone"] = Harvest::Timezones::MAPPING[tz]
+        self["timezone"] = if Harvest::Timezones::MAPPING[tz]
```
Comment (parndt) [inline lib/harvest/user.rb]: @zmoazeni was there any reason you're using `self["timezone"] = ` here instead of `self.timezone = ` as above?

Comment (jvdp) [inline lib/harvest/user.rb]: `self.timezone=` seems to be a recursive call. Also, this code now maps both ways and I don't think thats the intended behavior.

**Comment by jvdp** [comment]
+1 for dropping timezones if it's not needed. Does Harvest actually do anything with it?

**Comment by zmoazeni** [comment]
@parndt Yeah go ahead and sign up with a trial account which gives you full privileges. You can test it from that.

@jvdp Offhand, I don't think timezones are used for anything with the API.

**Comment by searls** [comment]
:+1: would love to kill this warning.

**Comment by jvdp** [comment]
Fixed by #86.

**Pull request closed without merging**